File: solutions/TypeScript/643.MaximumAverageSubarrayI.ts
function findMaxAverage(nums: number[], k: number): number {
  let windowSum: number = nums.slice(0, k).reduce((acc, num) => acc + num, 0);
  let maxSum: number = windowSum;

  for (let i: number = 1; i <= nums.length - k; i++) {
    windowSum = windowSum - nums[i - 1] + nums[i + k - 1];
    maxSum = Math.max(maxSum, windowSum);
  }

  return maxSum / k;
}

File: solutions/Python/643.MaximumAverageSubarrayI.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        window_sum: int = sum(nums[:k])
        sum_max: float = window_sum

        for i in range(1, len(nums) - k + 1):
            window_sum = window_sum - nums[i - 1] + nums[i + k - 1]
            sum_max = max(sum_max, window_sum)

        return sum_max / k
